tests/test-update-install-flatpaks.c

```diff
@@ -0,0 +1,2708 @@
+                                                     &error);
+
+
+  flatpak_build_dir = eos_test_get_flatpak_build_dir_for_updater_dir (updater_directory);
+  flatpak_repo_dir = g_file_get_child (flatpak_build_dir, "repo");
+  flatpak_apps_dir = g_file_get_child (flatpak_build_dir, "apps");
+  app_dir = g_file_get_child (flatpak_apps_dir, flatpaks_to_install[0].app_id);
+  app_dir_path = g_file_get_path (app_dir);
+  flatpak_repo_path = g_file_get_path (flatpak_repo_dir);
+  app_executable_path = g_build_filename (app_dir_path,
+                                          "files",
+                                          "bin",
+                                          "test",
+                                          NULL);
+
+  /* Get checksum for first installed flatpak */
+  initially_installed_flatpak_checksum =
+    get_checksum_for_flatpak_in_installation_dir (flatpak_user_installation_dir,
+                                                  "test-repo:app/org.test.Test",
+                                                  &error);
+
+  g_assert_no_error (error);
+
+  /* Slightly different contents so that the checksum will change */
+  g_file_set_contents (app_executable_path, "#!/bin/bash\nexit 1\n", -1, &error);
+  g_assert_no_error (error);
+
+  flatpak_build_export (updater_directory, app_dir_path, flatpak_repo_path, &error);
+  g_assert_no_error (error);
+
+  /* Update the server, so it has a new commit (1). */
+  etc_update_server (data, 1);
+  /* Update the client, so it also has a new commit (1) */
+  etc_update_client (data);
+
+  updated_flatpak_checksum =
+    get_checksum_for_flatpak_in_installation_dir (flatpak_user_installation_dir,
+                                                  "test-repo:app/org.test.Test",
+                                                  &error);
+
+  g_assert_cmpstr (initially_installed_flatpak_checksum, !=, updated_flatpak_checksum);
+}
+
+/* Insert a list of flatpaks to automatically install on the commit,
+ * then on the second commit, update the flatpak to the newest revision. The
+ * checksum for the flatpak pulled into the repo should differ on the second
+ * commit */
+static void
+test_update_flatpaks_updated_in_repo_after_install (EosUpdaterFixture *fixture,
+                                                    gconstpointer      user_data)
+{
+  g_auto(EtcData) real_data = { NULL, };
+  EtcData *data = &real_data;
+  FlatpakToInstall flatpaks_to_install[] = {
+    { "install", "com.endlessm.TestInstallFlatpaksCollection", "test-repo", "org.test.Test", "app", FLATPAK_TO_INSTALL_FLAGS_NONE }
+  };
+  FlatpakToInstall flatpaks_to_install_on_second_commit[] = {
+    { "install", "com.endlessm.TestInstallFlatpaksCollection", "test-repo", "org.test.Test", "app", FLATPAK_TO_INSTALL_FLAGS_NONE },
+    { "update", "com.endlessm.TestInstallFlatpaksCollection", "test-repo", "org.test.Test", "app", FLATPAK_TO_INSTALL_FLAGS_NONE }
```

Comment:
Something’s not working as expected with this test: if I comment out the `update` entry in `flatpaks_to_install_on_second_commit`, the test still passes. @smspillaz, can you take a look please? I have not managed to get my head around all the state going on in these tests.

Reply:
(This one still needs looking at, @smspillaz, in my rebased version.)

Reply:
Okay, I'll make sure to have a look - good catch.